docs/PythonAPI.md

```diff
@@ -0,0 +1,867 @@
+class DeviceIdentifier:
+    id: Tuple[int, int]
+    kind: DeviceKind
+    name: str
+
+    def __init__(self, *args: None, **kwargs: Any) -> None: ...
+    """Initializes a DeviceIdentifier object with optional arguments."""
+
+    def dict(self) -> Dict[str, Any]: ...
+    """Returns a dictionary of the DeviceIdentifier attributes."""
+
+    @staticmethod
+    def none() -> DeviceIdentifier: ...
+    """Returns a "None" type DeviceIdentifier. Useful when a DeviceIdentifier is not needed."""
+
+    def __eq__(self, other: object) -> bool:
+        """Checks if two DeviceIdentifier objects are equal."""
+        
+    def __ge__(self, other: object) -> bool:
+        """Checks if this DeviceIdentifier is greater than or equal to another."""
+        
+    def __gt__(self, other: object) -> bool:
+        """Checks if this DeviceIdentifier is greater than another."""
+        
+    def __le__(self, other: object) -> bool:
+        """Checks if this DeviceIdentifier is less than or equal to another."""
+        
+    def __lt__(self, other: object) -> bool:
+        """Checks if this DeviceIdentifier is less than another."""
+        
+    def __ne__(self, other: object) -> bool:
+        """Checks if two DeviceIdentifier objects are not equal."""
+```
+
+- `id`: A tuple containing two integers that serve as the device ID.
+
+- `kind`: An instance of the `DeviceKind` class that represents the type or kind of the device.
+
+- `name`: A string representing the name or label of the device.
+
+- The `dict` method creates a dictionary of a `DeviceIdentifier` object's attributes.
+
+## Class `DeviceKind`
+
+The `DeviceKind` class represents properties for supported devices, such as a camera or video source, in a given system.
+
+```python
+class DeviceKind:
+    Camera: ClassVar[DeviceKind] = DeviceKind.Camera
+    NONE: ClassVar[DeviceKind] = DeviceKind.NONE
+    Signals: ClassVar[DeviceKind] = DeviceKind.Signals
+    StageAxis: ClassVar[DeviceKind] = DeviceKind.StageAxis
+    Storage: ClassVar[DeviceKind] = DeviceKind.Storage
+
+    def __init__(self, *args: None, **kwargs: Any) -> None:
+        """Initializes the DeviceKind class."""
+
+    def __eq__(self, other: object) -> bool:
+        """Checks if two DeviceKind objects are equal."""
+        
+    def __ge__(self, other: object) -> bool:
+        """Checks if this DeviceKind is greater than or equal to another."""
+        
+    def __gt__(self, other: object) -> bool:
+        """Checks if this DeviceKind is greater than another."""
+        
+    def __int__(self) -> int:
+        """Converts the DeviceKind to an integer."""
+        
+    def __le__(self, other: object) -> bool:
+        """Checks if this DeviceKind is less than or equal to another."""
+        
+    def __lt__(self, other: object) -> bool:
+        """Checks if this DeviceKind is less than another."""
+        
+    def __ne__(self, other: object) -> bool:
+        """Checks if two DeviceKind objects are not equal."""
```

Comment:
Same thing here, it's probably enough to say that the `DeviceKind` can be ordered and compared.

Reply:
Noted. I'm going to leave this as-is for now in case we move forward with incorporating these as doc string in the pyi file.